File: src/main/java/com/aladdinworks33/service/MaintenanceTaskService.java
package com.aladdinworks33.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks33.domain.MaintenanceTask;
import com.aladdinworks33.dto.MaintenanceTaskDTO;
import com.aladdinworks33.dto.MaintenanceTaskSearchDTO;
import com.aladdinworks33.dto.MaintenanceTaskPageDTO;
import com.aladdinworks33.dto.MaintenanceTaskConvertCriteriaDTO;
import com.aladdinworks33.service.GenericService;
import com.aladdinworks33.dto.common.RequestDTO;
import com.aladdinworks33.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface MaintenanceTaskService extends GenericService<MaintenanceTask, Integer> {

	List<MaintenanceTask> findAll();

	ResultDTO addMaintenanceTask(MaintenanceTaskDTO maintenanceTaskDTO, RequestDTO requestDTO);

	ResultDTO updateMaintenanceTask(MaintenanceTaskDTO maintenanceTaskDTO, RequestDTO requestDTO);

    Page<MaintenanceTask> getAllMaintenanceTasks(Pageable pageable);

    Page<MaintenanceTask> getAllMaintenanceTasks(Specification<MaintenanceTask> spec, Pageable pageable);

	ResponseEntity<MaintenanceTaskPageDTO> getMaintenanceTasks(MaintenanceTaskSearchDTO maintenanceTaskSearchDTO);
	
	List<MaintenanceTaskDTO> convertMaintenanceTasksToMaintenanceTaskDTOs(List<MaintenanceTask> maintenanceTasks, MaintenanceTaskConvertCriteriaDTO convertCriteria);

	MaintenanceTaskDTO getMaintenanceTaskDTOById(Integer maintenanceTaskId);







}

File: aladdinworks33_ui/src/services/MaintenanceTaskService.js
import http from "../http-common"; 

class MaintenanceTaskService {
  getAllMaintenanceTasks(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/maintenanceTask/maintenanceTasks`, searchDTO);
  }

  get(maintenanceTaskId) {
    return this.getRequest(`/maintenanceTask/${maintenanceTaskId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/maintenanceTask?field=${matchData}`, null);
  }

  addMaintenanceTask(data) {
    return http.post("/maintenanceTask/addMaintenanceTask", data);
  }

  update(data) {
  	return http.post("/maintenanceTask/updateMaintenanceTask", data);
  }
  
  uploadImage(data,maintenanceTaskId) {
  	return http.postForm("/maintenanceTask/uploadImage/"+maintenanceTaskId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new MaintenanceTaskService();
